### DISCUSSION TITLE
Question about using multiple dialogs

### DISCUSSION THREAD

**Discussion by ardura** [Barugon/egui_file#46] [Q&A] [answered]
I want to have different save/load dialogs for different things in my application. Is there a way I can sync the directory between different instances?

I'm trying to fix navigating to the same directory to save that I opened from in different dialogs. I've been trying to figure this out but haven't been able to due to the PathBuf struct used in egui_file not allowing copy, and some workarounds I tried like Arc + Mutex have made things way more complex than they probably need to be. I'm curious if anyone else has gotten something like this working.

**Comment by Barugon** [comment, marked as answer]
This might be accomplished by listening to directory changed events but I'm not sure how cross-platform that would be.

[edit] This looks promising: https://github.com/notify-rs/notify

**Reply by ardura** [reply]
I think that's a good solution, however I think I'll just deal with having to navigate once for open and once for save so it remains cross platform. Great crate!
